packages/main/src/plugin/context-registry.ts

```diff
@@ -28,13 +28,21 @@ export class ContextRegistry {
 
   registerContext(extension: string): void {
     if (!this.contexts.find(context => context.extension === extension)) {
-      const ctxWithLargestId = this.contexts.reduce((prev, current) => (prev.id > current.id ? prev : current));
-      const newId = ctxWithLargestId.id + 1;
-      const ctx = new Context(newId, null, extension, this.apiSender);
+      const id = this.getNewId();
+      const ctx = new Context(id, null, extension, this.apiSender);
       this.contexts.push(ctx);
     }
   }
 
+  getNewId() {
+    let newId = 0;
```

Comment:
just a question : do we need to have this code or we could just increment the counter ?

Reply:
no you're right, done!!

Reply:
it was just a question, but yes if we can keep things simple